solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/SelectStream.java

```diff
@@ -52,14 +53,21 @@ public class SelectStream extends TupleStream implements Expressible {
   private TupleStream stream;
   private StreamContext streamContext;
   private Map<String, String> selectedFields;
+  private List<String> selectedFieldGlobPatterns;
   private Map<StreamEvaluator, String> selectedEvaluators;
   private List<StreamOperation> operations;
 
   public SelectStream(TupleStream stream, List<String> selectedFields) throws IOException {
     this.stream = stream;
     this.selectedFields = new HashMap<>();
+    this.selectedFieldGlobPatterns = new ArrayList<>();
     for (String selectedField : selectedFields) {
-      this.selectedFields.put(selectedField, selectedField);
+      if (selectedField.contains("*")) {
```

Comment:
Can a glob go anywhere in a field name? i.e is `*_a` or `a_*_b` valid? Or is that not for us to determine here?

Reply:
Yup, a glob can go anywhere, so any of those options would be valid.